rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -125,11 +134,15 @@ def subscribe(self, client_id, callback):
 
         """
         with self.lock:
-            self.subscriptions[client_id] = callback
-            # If the topic is latched, add_callback will immediately invoke
+            # If the topic is latched, adding a new subscriber will immediately invoke
             # the given callback.
-            self.subscriber.impl.add_callback(self.callback, [callback])
-            self.subscriber.impl.remove_callback(self.callback, [callback])
+            # In any case, the first message is handled using new_sub_callback,
+            # which adds the new callback to the subscriptions dictionary.
+            qos_profile = QoSProfile(durability=QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL)
+            self.new_subscriptions.update({client_id: callback})
+            if self.new_subscriber is None:
+                self.new_subscriber = self.node_handle.create_subscription(
+                    self.msg_class, self.topic, self._new_sub_callback, qos_profile=qos_profile)
```

Comment:
It is indeed a bit of a workaround; the point is to get a (potentially) 'latched' message from a topic with an existing subscriber.

Using one subscriber per client id could work as well for that case, but in the case the topic not latched and a publisher is sending data at a high rate it would be worse than this solution IMO.